### PR TITLE
Use GPU more efficiently

### PR DESCRIPTION
**Pull request by gennarinoos**
- Move edge predictor and face classifier computation to GPU only
- Loss calculation on GPU
- Rest happens on CPU

### CODE REVIEW

File: src/neural_mesh_simplification/losses/combined_loss.py [hunk 1]
```diff
@@ -26,29 +32,40 @@ def __init__(
...
-        edge_crossing_loss = self.edge_crossing_loss(simplified_data)
-        overlapping_triangles_loss = self.overlapping_triangles_loss(simplified_data)
+        edge_crossing_loss = self.edge_crossing_loss(sampled_vertices, sampled_faces, face_probs)
+
+        del face_probs
+
+        overlapping_triangles_loss = self.overlapping_triangles_loss(sampled_vertices, sampled_faces)
+
+        del sampled_vertices
+        del sampled_faces
```
Comment (martinnormark) [inline src/neural_mesh_simplification/losses/combined_loss.py:68]: Isn't it safe to delete these as well?

```suggestion
        del sampled_vertices
        del sampled_faces
        del original_x
        del original_face

```

File: src/neural_mesh_simplification/losses/combined_loss.py [hunk 2]
```diff
@@ -11,13 +12,18 @@
...
-        self.prob_chamfer_loss = ProbabilisticChamferDistanceLoss()
-        self.prob_surface_loss = ProbabilisticSurfaceDistanceLoss()
-        self.collision_loss = TriangleCollisionLoss()
-        self.edge_crossing_loss = EdgeCrossingLoss()
+        self.device = device
+        self.prob_chamfer_loss = ProbabilisticChamferDistanceLoss().to(self.device)
+        self.prob_surface_loss = ProbabilisticSurfaceDistanceLoss().to(self.device)
+        self.collision_loss = TriangleCollisionLoss().to(self.device)
+        self.edge_crossing_loss = EdgeCrossingLoss().to(self.device)
         self.overlapping_triangles_loss = OverlappingTrianglesLoss()
```
Comment (martinnormark) [inline src/neural_mesh_simplification/losses/combined_loss.py:27]: Is it intentional not to move this loss function to the same device?

```suggestion
        self.overlapping_triangles_loss = OverlappingTrianglesLoss().to(self.device)
```

**Review by martinnormark — APPROVED** [review]
Awesome 💪🏻

A few questions with suggestions.

File: src/neural_mesh_simplification/losses/combined_loss.py (hunk 2 above)
Comment (gennarinoos) [reply src/neural_mesh_simplification/losses/combined_loss.py:27]: It's an oversight. Good catch!
